Composer/packages/ui-plugins/cross-trained/src/index.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { PluginConfig } from '@bfc/extension';
+import { SDKKinds } from '@bfc/shared';
+
+const config: PluginConfig = {
+  recognizers: [
+    {
+      id: SDKKinds.CrossTrainedRecognizerSet,
+      displayName: 'Default recognizer',
+      isSelected: (data) => {
+        return typeof data === 'string';
+      },
+      handleRecognizerChange: (props, shellData, _) => {
+        const { qnaFiles, luFiles, currentDialog, locale } = shellData;
+        const qnaFile = qnaFiles.find((f) => f.id === `${currentDialog.id}.${locale}`);
+        const luFile = luFiles.find((f) => f.id === `${currentDialog.id}.${locale}`);
+
+        if (qnaFile && luFile) {
+          // strip locale out of id so it doesn't get serialized
+          // into the .dialog file
+          props.onChange(`${currentDialog.id}.lu.qna`);
```

Comment:
Duplication. Is it necessary? what's the relationship between `handleRecognizerChange()` and the useEffect hook in `RecognizerField.tsx`?
![image](https://user-images.githubusercontent.com/8528761/90367756-bcc80800-e09b-11ea-9b24-5d2fdb6f53b4.png)

Reply:
useEffect logic is for migrating with old bot whose recognizer is undefined.

Reply:
It will be burden to maintain it in the future. we can share something between `ui-plugins` and `extensions`.

The shared logic here is, given `qnaFiles` and `luFiles`, generates a recognizer field value (now it's a plain string, soon it will become an object with explict recognizer $kind)

Reply:
This is really important. I think the naming of qna file and recognizer filed value are both evolving.